lib/ansible/plugins/filter/ipaddr.py

```diff
@@ -982,10 +982,10 @@ def _need_netaddr(f_name, *args, **kwargs):
                                     'installed on the ansible controller' % f_name)
 
 
-def ip4_hex(arg):
+def ip4_hex(arg, delimiter=''):
     ''' Convert an IPv4 address to Hexadecimal notation '''
     numbers = list(map(int, arg.split('.')))
-    return '{:02x}{:02x}{:02x}{:02x}'.format(*numbers)
+    return '{:02x}{sep}{:02x}{sep}{:02x}{sep}{:02x}'.format(*numbers, sep=delimiter)
```

Comment:
This is not valid in Python 2.6. While I know the original is also invalid, I'd rather see

`return '{0:02x}{sep}{1:02x}{sep}{2:02x}{sep}{3:02x}'.format(*numbers, sep=delimiter)`

Reply:
I have updated my commit and implemented your format fix.